src/isomorphism/vf2.rs

```diff
@@ -318,51 +335,20 @@ where
     }
 }
 
-fn reindex_graph<Ty>(py: Python, graph: &StablePyGraph<Ty>) -> StablePyGraph<Ty>
-where
-    Ty: EdgeType,
-{
-    // NOTE: this is a hacky workaround to handle non-contiguous node ids in
-    // VF2. The code which was forked from petgraph was written assuming the
-    // Graph type and not StableGraph so it makes an implicit assumption on
-    // node_bound() == node_count() which isn't true with removals on
-    // StableGraph. This compacts the node ids as a workaround until VF2State
-    // and try_match can be rewitten to handle this (and likely contributed
-    // upstream to petgraph too).
-    let mut new_graph = StablePyGraph::<Ty>::default();
-    let mut id_map: HashMap<NodeIndex, NodeIndex> = HashMap::new();
-    for node_index in graph.node_indices() {
-        let node_data = graph.node_weight(node_index).unwrap();
-        let new_index = new_graph.add_node(node_data.clone_ref(py));
-        id_map.insert(node_index, new_index);
-    }
-    for edge in graph.edge_references() {
-        let edge_w = edge.weight();
-        let p_index = id_map[&edge.source()];
-        let c_index = id_map[&edge.target()];
-        new_graph.add_edge(p_index, c_index, edge_w.clone_ref(py));
-    }
-
-    new_graph
-}
-
 trait SemanticMatcher<T> {
     fn enabled(&self) -> bool;
-    fn eq(&mut self, _: &T, _: &T) -> PyResult<bool>;
+    fn eq(&mut self, _: Python, _: &T, _: &T) -> PyResult<bool>;
 }
 
-impl<T, F> SemanticMatcher<T> for Option<F>
-where
-    F: FnMut(&T, &T) -> PyResult<bool>,
-{
+impl SemanticMatcher<PyObject> for Option<PyObject> {
     #[inline]
     fn enabled(&self) -> bool {
         self.is_some()
     }
     #[inline]
-    fn eq(&mut self, a: &T, b: &T) -> PyResult<bool> {
-        let res = (self.as_mut().unwrap())(a, b)?;
-        Ok(res)
+    fn eq(&mut self, py: Python, a: &PyObject, b: &PyObject) -> PyResult<bool> {
+        let res = self.as_mut().unwrap().call1(py, (a, b))?;
+        res.is_true(py)
     }
 }
```

Comment:
It's not really related to this PR, but I'm wondering if this makes sense to just refactor to not be a trait? It seems we don't this to be generic over any type `T` that's a potential weight as we always are using a `PyObject`. This was originally forked from petgraph and they need the generic definition here because the weight in a graph type in petgraph can be any type. But that's something we can do in a followup, it's not worth changing here.

Reply:
Actually i like this even if we don't really use it but yeah, we can remove it in a followup PR